[RFR] Test cases for eap7, eapxp, quarkus1 ruleset analysis

nitishSr opened the pull request:
- Added test case for running simple analysis on a project in IntelliJ IDE
- Added logic to verify story points for analysis on report page for eclipse, code ready studio and IntelliJ IDEs

Attached test run recording and result for tests
![test_result_ide](https://user-images.githubusercontent.com/43022982/124929708-57b43280-e01e-11eb-8680-40042425451c.png)
- 

https://user-images.githubusercontent.com/43022982/124751809-aee5d480-df44-11eb-81ca-a729a9d2b572.mp4


**NOTE - There is an issue with target field, it selects the target from previous configuration automatically, which is not ideal**

nitishSr committed:
Test case for IntelliJ mta run analysis

nitishSr committed:
Added steps to perform simple analysis for a project in IntelliJ

nitishSr committed:
Added snapshots for intellij and eclipse/codeready studio report verification

nitishSr committed:
Updated READ ME with IntelliJ inclusion

nitishSr committed:
Added logic for story points verification for eclipse, codeready studio and intellij

sshveta commented on src/fixtures/ide_fixture.py:
```diff
@@ -33,6 +34,9 @@ def setup_codereadystudio(config):
     codereadystudio.set_default_timeout(timeout=config["timeout_in_seconds"])
     yield codereadystudio
     # codereadystudio.close_application(codereadystudio_app)
```
we can remove this comment now

nitishSr replied:
Sure

sshveta commented on src/lib/application.py:
```diff
@@ -420,3 +456,126 @@ def is_analysis_complete(self):
+            self.click_element(locator_type="image", locator="mta_cli_input.png")
+            self.type_text(config_data["mta_cli_path"])
+        except Exception as exc:
+            logging.debug("MTA cli path is already present ! {}".format(exc))
```
The test should not fail if the cli path is already present .

nitishSr replied:
Yes, this except block ensures, test does not fail, if cli path is already present

sshveta commented on src/lib/application.py:
```diff
@@ -294,10 +314,26 @@ def is_analysis_complete(self):
+            (bool): True if story points were accurate
+        """
+        try:
+            self.wait_find_element(locator_type="image", locator="story_points.png")
```
It is checking for story point image right ? How can we validate if the story point is a number and same for all IDE's

nitishSr replied:
Yes, this is specifically, checking for acemair project, which generates 2 story points. It is being validated against image (that has 2 story points from web report) So, we need not worry about it being a number, if story points are generated other than 2, test will fail. Overall, this validation will always apply right check. And yes, this is same for all IDEs, since project is same

sshveta reviewed (changes requested):
Looks Good @nitishSr .
Some questions 
1) Are all IDE's using same application for analysis ?
2) If not then they should and  same story points should generate for all of them.
3) I want to test one quarkus1 source mode application too as that is more accurate test and latest is thorntail.can we find a way to pass target and app and story points . Meaning we will pass these three things in an array [app, target , story point] and it will run for eap7 (current app) , quarkus 1 and eapxp

nitishSr commented:
> Looks Good @nitishSr .
> Some questions
> 
> 1. Are all IDE's using same application for analysis ?
> 2. If not then they should and  same story points should generate for all of them.
> 3. I want to test one quarkus1 source mode application too as that is more accurate test and latest is thorntail.can we find a way to pass target and app and story points . Meaning we will pass these three things in an array [app, target , story point] and it will run for eap7 (current app) , quarkus 1 and eapxp

1) Yes, all are using same application, whose path is provided in the config json file
2) All are being validated against same story points
3) Need to add target param for Eclipse and VS Code IDEs, those are missing (implemented for IntelliJ). Will create 2 new test cases, where we will verify the quarkus1 application and eapxp separately. Hope it will cover these points.

Will make the changes and try to cover these migrations atleast.

nitishSr committed:
Added new test cases for coverage of eapxp, quarkus rulesets and story points verification

nitishSr committed:
Added eapxp and quarkus1 analysis tests for IntelliJ

nitishSr commented:
> > Looks Good @nitishSr .
> > Some questions
> > 
> > 1. Are all IDE's using same application for analysis ?
> > 2. If not then they should and  same story points should generate for all of them.
> > 3. I want to test one quarkus1 source mode application too as that is more accurate test and latest is thorntail.can we find a way to pass target and app and story points . Meaning we will pass these three things in an array [app, target , story point] and it will run for eap7 (current app) , quarkus 1 and eapxp
> 
> 1. Yes, all are using same application, whose path is provided in the config json file
> 2. All are being validated against same story points
> 3. Need to add target param for Eclipse and VS Code IDEs, those are missing (implemented for IntelliJ). Will create 2 new test cases, where we will verify the quarkus1 application and eapxp separately. Hope it will cover these points.
> 
> Will make the changes and try to cover these migrations atleast.

@sshveta , the changes have been made. Added test cases for eap7, eapxp and quarkus1 rulesets for all the IDEs. There are some changes in Eclipse CHE workspace setup, hence will cover that in a new PR, as need to change web locators. For now, as discussed, I have modified and added test cases for Eclipse, Code ready studio, VS Code and IntelliJ. For future purpose, if we get new rulesets, we simply need to add that test case, and remaining logic will take care of analysis based on target we pass.
![vscode_test_results](https://user-images.githubusercontent.com/43022982/127835154-07fe86d1-f359-41cd-be5c-530c2511d8e0.png)
![code_ready_studio_test_result](https://user-images.githubusercontent.com/43022982/127835173-26a171bf-7c25-41a5-bec1-d8a024d93efb.png)
![eclipse_test_results](https://user-images.githubusercontent.com/43022982/127835184-1819b5b2-6925-45e0-9933-aceb97177a3b.png)
